### PR TITLE
Added animated gif support

### PR DESCRIPTION
**Pull request by eadmaster**
Added animated gif  support via [this lib](https://github.com/bitbank2/AnimatedGIF/):

 - Gif files can be viewed in the file manager
 - show `boot.gif` during boot if present in any storage (press any button to stop the animation)

### CODE REVIEW

**Comment by eadmaster** [comment]
UPDATE: feature restricted to the cardputer only, on `m5stack-cplus2` build fails due to `dram0_0_seg' overflowed by 17712 bytes, i guess it is the same on other non-S3 boards 

https://github.com/pr3y/Bruce/actions/runs/11268950675/job/31336615019?pr=328

 uncomment this line to test on other boards:

https://github.com/pr3y/Bruce/blob/ebe45f959ff8ede696032fd0b49193470c8f5273/src/core/display.cpp#L1008

File: src/core/display.cpp [hunk 1]
```diff
@@ -820,6 +819,218 @@ bool showJpeg(FS fs, String filename, int x, int y) {
...
+#include <AnimatedGIF.h>
+
+#define NORMAL_SPEED
+//#define USE_DMA
+
+#ifdef USE_DMA
+  uint16_t usTemp[2][SAFE_STACK_BUFFER_SIZE]; // Global to support DMA use
+#else
+  uint16_t usTemp[1][SAFE_STACK_BUFFER_SIZE];    // Global to support DMA use
+#endif
```
Comment (bmorcelli) [inline src/core/display.cpp:837]: These are the villains of DRAM memory..

Isn't it possible to use malloc to address it into heap memory, and use the funtions to point into this address?

**Comment by Ademolght** [comment]
@eadmaster awesome

File: src/core/display.cpp (hunk 1 above)
Comment (eadmaster) [reply src/core/display.cpp:837]: just tried , still seem to overflow dram0_0_seg...

**Comment by LEGOATDU49** [comment]
The gif dosent work for me on Cardputer, I have resize in 240x135 and I have download the latest version on GitHub
ParkourGif

**Comment by bmorcelli** [comment]
Available on Beta, for now

**Comment by LEGOATDU49** [comment]
The beta is it on the web flasher

**Comment by bmorcelli** [comment]
No, because it has an error on Cores3 device

**Comment by eadmaster** [comment]
> No, because it has an error on Cores3 device

i was able to reproduce this error, it seems related to this line:
https://github.com/pr3y/Bruce/blob/main/src/core/display.cpp#L896

Looks a compatibility issue with the lgfx library, fixed by adding a preprocessor directive around these `GIF*` functions:
https://github.com/pr3y/Bruce/commit/e9a4b8b85cbf1b032ef4f02520924270e0237d5a

**Comment by bmorcelli** [comment]
I just patched this line to allow the beta files to be uploaded

**Comment by eadmaster** [comment]
ok, it's passing now

**Pull request merged**